HW2/P2/parallel_vector.pyx

```diff
@@ -109,7 +111,16 @@ cpdef move_data_medium_grained(np.int32_t[:] counts,
        for r in range(repeat):
            for idx in range(src.shape[0]):
                if counts[src[idx]] > 0:
```

Comment:
what if counts[src[idx]] changes between now and acquiring the lock? You usually want to check the same condition again after acquiring the lock